### PR TITLE
Add cargo-spellcheck

### PR DESCRIPTION
**Pull request by jayvdb**
https://github.com/drahnr/cargo-spellcheck

### CODE REVIEW

**Comment by jayvdb** [comment]
@taiki-e, this requires GLIBC_2.34 and GLIBCXX_3.4.29, not available on older Linux.

I don't see a mechanism to indicate that a binary doesnt support older Linux.  I'm happy to add something like that so that the CI can skip ones where it isnt supported.  Sounds like a fairly large chunk of work though.

**Comment by taiki-e** [comment]
> I don't see a mechanism to indicate that a binary doesnt support older Linux.

Please update one of the `*_incompat` arrays in `tools/ci/tool-list.sh`:

https://github.com/taiki-e/install-action/blob/main/DEVELOPMENT.md#troubleshooting

> If CI fails only for containers using older versions of glibc or musl, you may need to add the tool name to one of the `*_incompat` arrays in `tools/ci/tool-list.sh`.

File: README.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ https://spdx.org/licenses
...
 | [**cargo-rdme**](https://github.com/orium/cargo-rdme) | `$CARGO_HOME/bin` | [GitHub Releases](https://github.com/orium/cargo-rdme/releases) | Linux, macOS, Windows | [MPL-2.0](https://github.com/orium/cargo-rdme/blob/HEAD/LICENSE.md) |
 | [**cargo-sort**](https://github.com/DevinR528/cargo-sort) | `$CARGO_HOME/bin` | [GitHub Releases](https://github.com/DevinR528/cargo-sort/releases) | Linux, macOS, Windows | [Apache-2.0 OR MIT](https://github.com/DevinR528/cargo-sort/blob/55ec89082466f6bb246d870a8d56d166a8e1f08b/Cargo.toml#L5) |
+| [**cargo-spellcheck**](https://github.com/drahnr/cargo-spellcheck) | `$CARGO_HOME/bin` | [GitHub Releases](https://github.com/drahnr/cargo-spellcheck/releases) | Linux and Windows | [LGPLv2.1](https://github.com/drahnr/cargo-spellcheck/blob/f60c81ad0134c12faf08715f08d14f66e749f6e7/docs/checkers.md#L38) |
```
Comment (taiki-e) [inline README.md:100]:
```suggestion
| [**cargo-spellcheck**](https://github.com/drahnr/cargo-spellcheck) | `$CARGO_HOME/bin` | [GitHub Releases](https://github.com/drahnr/cargo-spellcheck/releases) | Linux and Windows | [LGPLv2.1](https://github.com/drahnr/cargo-spellcheck/blob/f60c81ad0134c12faf08715f08d14f66e749f6e7/docs/checkers.md#nlprules) |
```
(GitHub uses preview mode for markdown by default, so L* doesn't work without additional action or `?plain=1`)

**Review by taiki-e — APPROVED** [review]
Thanks.

**Pull request merged**